blocks/library/html/index.js

```diff
@@ -41,60 +41,42 @@ registerBlockType( 'core/html', {
 		},
 	},
 
-	edit: class extends Component {
-		constructor() {
-			super( ...arguments );
-			this.preview = this.preview.bind( this );
-			this.edit = this.edit.bind( this );
-			this.state = {
-				preview: false,
-			};
-		}
-
-		preview() {
-			this.setState( { preview: true } );
-		}
-
-		edit() {
-			this.setState( { preview: false } );
-		}
-
-		render() {
-			const { preview } = this.state;
-			const { attributes, setAttributes, focus } = this.props;
-
-			return (
-				<div>
-					{ focus &&
-						<BlockControls key="controls">
-							<div className="components-toolbar">
-								<button className={ `components-tab-button ${ ! preview ? 'is-active' : '' }` } onClick={ this.edit }>
-									<span>HTML</span>
-								</button>
-								<button className={ `components-tab-button ${ preview ? 'is-active' : '' }` } onClick={ this.preview }>
-									<span>{ __( 'Preview' ) }</span>
-								</button>
-							</div>
-						</BlockControls>
-					}
-					{ preview ?
-						<div dangerouslySetInnerHTML={ { __html: attributes.content } } /> :
-						<TextareaAutosize
-							value={ attributes.content }
-							onChange={ ( event ) => setAttributes( { content: event.target.value } ) }
-						/>
-					}
-					{ focus &&
-						<InspectorControls key="inspector">
-							<BlockDescription>
-								<p>{ __( 'Add custom HTML code and preview it right here in the editor.' ) }</p>
-							</BlockDescription>
-						</InspectorControls>
-					}
+	edit: withState( {
+		preview: false,
+	} )( ( { attributes, setAttributes, setState, focus, preview } ) => [
```

Comment:
Should we start using `React.Fragment`? Do we want to update Babel first? :)

Reply:
>Should we start using `React.Fragment` to avoid using keys? Do we want to update Babel first? :)

Yeah, I'd probably want to avoid the direct reference to `React.Fragment`, but if we had `<>` syntax available, I'd be open to using it. Maybe in a subsequent pull request where we upgrade Babel?

Reply:
Sure, I can spin up PR with Babel upgrade on Monday :)